lib/src/helpers/authentication-helper.ts

```diff
@@ -0,0 +1,233 @@
+            credentials: "include",
+            headers: {
+                "Content-Type": "application/x-www-form-urlencoded",
+                authorization: `Bearer ${accessToken}`
+            },
+            method: "POST",
+            mode: "cors"
+        };
+
+        try {
+            const res = await fetch(config?.stsTokenEndpoint ?? "", requestOptions);
+            if (!res.ok) {
+                throw new AsgardeoAuthException(
+                    "TOKEN_EXCHANGE-AUTH_HELPER-EAT1-NE01",
+                    "Invalid exchanging token response",
+                    "Token exchange grant has been failed"
+                );
+            }
+
+            const responseBody: StsExchangeResponse = await res.json();
+
+            if (responseBody?.access_token) {
+                await this._dataLayer.setCustomData(
+                    StsStore.SessionData,
+                    responseBody
+                );
+            }
+
+        } catch (error) {
+            throw new AsgardeoAuthException(
+                "TOKEN_EXCHANGE-AUTH_HELPER-EAT1-NE02",
+                "Error in exchanging token",
+                "Token exchange grant has been failed"
+            );
+        }
+    }
+
+    public async requestAccessToken(
+        authorizationCode?: string,
+        sessionState?: string,
+        checkSession?: () => Promise<void>,
+        pkce?: string,
+        state?: string
+    ): Promise<BasicUserInfo> {
+        try {
+            const userInfo = await super.requestAccessToken(
+                authorizationCode,
+                sessionState,
+                checkSession,
+                pkce,
+                state
+            );
+
+            await this.exchangeAccessToken();
+
+            return userInfo;
+        } catch (error: any) {
+            return Promise.reject(
+                new AsgardeoAuthException(
+                    "TOKEN_EXCHANGE-AUTH_HELPER-RAT3-NF01",
+                    error?.name ?? "Failed requesting access token.",
+                    error?.message ?? "Failed retrieving user info."
+                )
+            );
+        }
+    }
+
+    public async getAccessToken(): Promise<string> {
+        const stsSessionData: StsExchangeResponse =
+            await this._dataLayer.getCustomData<StsExchangeResponse>(
+                StsStore.SessionData
+            );
+
+        return stsSessionData?.access_token;
+    }
+
+    public async refreshAccessToken(
+        enableRetrievingSignOutURLFromSession?: (
+            config: SPACustomGrantConfig
+        ) => void
+    ): Promise<BasicUserInfo> {
+        const userInfo = await super.refreshAccessToken(
+            enableRetrievingSignOutURLFromSession
+        );
+
+        await this.exchangeAccessToken();
+
+        return userInfo;
+    }
+
+    public async getDecodedIDToken(): Promise<DecodedIDTokenPayload> {
+        // TODO Uncomment this once choreo STS enable id token in the response. 
```

Comment:
Do we need this comment?

Reply:
Yes. Choreo STS is not returning an ID token at the moment. Once they enable the id token, we can remove the comment and refer to their id token over Asgardeo id token.